react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift

```diff
@@ -188,18 +204,28 @@ public class RNReactNativeGutenbergBridge: RCTEventEmitter {
         }
     }
 
-    private func shouldLog(with level: Int) -> Bool {
-        return level >= RCTGetLogThreshold().rawValue
+    public func connectionEstablished() {
+        guard !hasObservers else { return } // We have an established connection no need to continue.
+        hasObservers = true
+        while (self.queuedEvents.count > 0) {
+            let event = self.queuedEvents.removeFirst()
+            super.sendEvent(withName: event.name, body: event.body) // execute this on super as we want to avoid logic in self.
+        }
     }
 
-    override public func startObserving() {
-        super.startObserving()
-        hasObservers = true
+    public override func sendEvent(withName name: String!, body: Any!) {
+        DispatchQueue.main.async {
+            if self.hasObservers && self.queuedEvents.count == 0 {
+                super.sendEvent(withName: name, body: body)
+            } else {
+                let event = GutenbergEvent(name: name, body: body)
```

Comment:
Sorry! I meant the warning on this line. I wrote the comment but it disappeared (second time that happens to me this week ¯\\\_(ツ)_/¯ )

The warning says `Coercion of implicitly unwrappable value of type 'Any?' to 'Any' does not unwrap optional`

Would the override still work if we change the optionality of the parameters?
Actually on a second look, that `Any!` looks quite dangerous. This might crash if any queued event had no body.

It would be great if `name` is not optional and `body` is an actual non-implicitly-unwrapped optional, but open to possibilities too.

Reply:
I just did a test and it doesn't crash. `Any` seems to accept a nil value even if it's not an optional itself (?)

But still... I wouldn't trust the runtime completely with this 😬

Reply:
> I just did a test and it doesn't crash. `Any` seems to accept a nil value even if it's not an optional itself (?)

Interesting! I wouldn't have expected that. Does that mean the struct's default initializer is not actually force-unwrapping the value? When you tested with a `nil`, do you know if the `GutenbergEvent` was initialized, or did the code branch directly to the `super.sendEvent`?

Reply:
> Perhaps there is some subtlety with permissiveness afforded for potential value or reference types? 🤔

This kinds of make sense in a way that `Optional` is basically a `Value type`, and we are passing `Optional<>.none` as `Any`. 

And `AnyObject` wouldn't accept it since it expects `Reference` types only, so it will try to unwrap the optional. It would probably not even compile.

Reply:
Thanks for the explanation Eduardo! I also found [this explanation](https://stackoverflow.com/a/34644253/882146), which is just as you mentioned. The discussion is interesting because it explains why you cannot directly assign nil to a var with type Any, since there is no inferred type for the generic in the Optional. That there is no error in the above case was surprising to me, but this explains it (kind of a language design edge case, I suppose).

Reply:
So @mkevins , what do you think: Should we avoid this edge case?

I think it would be good to try avoid it, but since it's clear now that it won't crash, maybe it's not a blocker for this PR. Still it would be good to clear the warning 😬

Reply:
> comment but it disappeared somehow

Oh, that is weird! Thanks for confirming though. 

>The warning says Coercion of implicitly unwrappable value of type 'Any?' to 'Any' does not unwrap optional

I totally missed this error thank you for catching it. Also thank you both for the investigation and knowledge share. That's really interesting on how `Any` and `Optionals` interact. 

> (kind of a language design edge case, I suppose).

Yeah this is neat, sometimes it's easy to forget that language design has its own edge cases to deal with.


> what do you think: Should we avoid this edge case?

To me, it seems like an easy fix to adjust. So I'll make that tweak today and push up a commit to this PR here shortly.

Reply:
This is actually an interesting situation. @etoledom I didn't run into the warning that you were seeing so I was wondering if it was a runtime flag, but I wasn't able to see it at runtime either. 

I do think it would be a good idea to be explicit about the optionality of the event parameters though so I did an update to address that. If you don't mind will you let me know if it fixes the warning?

XCode 11.5 
<img width="437" alt="Screen Shot 2020-06-17 at 12 38 24 PM" src="https://user-images.githubusercontent.com/3384451/84925502-fda1a180-b097-11ea-87cd-5ef932f9a355.png">

Reply:
Fwiw, I didn't see the warning either (but I was exploring the code via the WordPress-iOS project, so I wonder if that's a factor?)

Reply:
Oh right, you will see it if you run the gutenberg example project on Xcode.